### PR TITLE
Goodfet update - bugfixes

### PR DESCRIPTION
**Pull request by taylorcenters**

### CODE REVIEW

File: killerbee/__init__.py [hunk 1]
```diff
@@ -140,9 +175,6 @@ def __init__(self, device: str=None, datasource: str=None, gps: str=None) -> Non
                     if gfccspi and subtype == 0:
                         from .dev_telosb import TELOSB
                         self.driver = TELOSB(self.dev)
-                    elif gfccspi and subtype == 1:
```
Comment (rmspeers) [inline killerbee/__init__.py:143]: deprecating apimotev1? should that be in the README for devices or drivers or something if so?

File: killerbee/dev_template.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+            self.set_channel(channel, page)
+
+    def set_sync(self, sync: int=0xA70F) -> Any:
+        self.capabilities.require(KBCapabilities.SET_SYNC)
+
+        if self.handle is None:
+            raise Exception("Handle does not exist")
+
+    def jammer_off(self, channel: Optional[int]=None, page: int=0) -> None:
+        raise Exception('Not yet implemented')
```
Comment (rmspeers) [inline killerbee/dev_template.py:168]: Maybe put in `raise NotImplementedError()` type stuff in a lot of these so people know they have to replace and don't leave stub too much?

File: killerbee/dev_template.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    def jammer_on(self, channel: Optional[int]=None, page: int=0) -> None:
+        self.capabilities.require(KBCapabilities.PHYJAM_REFLEX)
+
+        if self.handle is None:
+            raise Exception("Handle does not exist")
+
+        if channel is None:
+            pass
+        else:
+            self.set_channel(channel, page)
```
Comment (rmspeers) [inline killerbee/dev_template.py:159]: Maybe add things like `# TODO: Enable jamming mode. Ensure that KBCapabilities checks align to the type of jamming that is being done.`

File: killerbee/dev_template.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    def set_channel(self, channel: int, page: int=0) -> None:
+        self.capabilities.require(KBCapabilities.SETCHAN)
+
+        if self.handle is None:
+            raise Exception("Handle does not exist")
+
+        if channel >= 11 and channel <= 26:
+            self._channel = channel
+        else:
+            raise Exception('Invalid channel')
```
Comment (rmspeers) [inline killerbee/dev_template.py:91]: My fault I'm sure, this should be probably a `ValueError`

File: killerbee/dev_template.py [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+        self.handle = None
+
+    def check_capability(self, capab: int) -> bool:
+        return self.capabilities.check(capab)
+
+    def get_capabilities(self) -> Dict[int, bool]:
+        return self.capabilities.getlist()
+
+    def get_dev_info(self) -> list[Union[str, Any]]:
+        return [self.dev, "GoodFET Apimote", ""]
```
Comment (rmspeers) [inline killerbee/dev_template.py:64]: Probably change Apimote stuff to be generic name or TODO all over this file.

File: killerbee/__init__.py (hunk 1 above)
Comment (taylorcenters) [reply killerbee/__init__.py:143]: Good catch

File: killerbee/dev_template.py [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+
+class APIMOTE:
+    def __init__(self, dev: str) -> None:
+
+        self._channel: Optional[int] = None
+        self._page: int = 0
+        self.handle: Optional[Any] = None
+        self.dev: str = dev
+
+        self.handle = GoodFETCCSPI()
```
Comment (rmspeers) [inline killerbee/dev_template.py:22]: Few things probably are going to heavily rely on another thing like the import on line 12 and 22 - likely it's a usb handle here that they'd generally keep - may be a good example to set up like that?

Comment (taylorcenters) [reply killerbee/dev_template.py:22]: will make further developer instruction changes in a different PR

File: killerbee/dev_template.py (hunk 5 above)
Comment (taylorcenters) [reply killerbee/dev_template.py:64]: noted. thanks

File: killerbee/dev_template.py (hunk 4 above)
Comment (taylorcenters) [reply killerbee/dev_template.py:91]: lets chat about what this should look like sometime

**Review by jspiel — APPROVED** [review]

**Pull request merged**